### PR TITLE
[elm] Import types in operations

### LINKED ISSUE #2484: [BUG] [Elm] Uuid import and encoding missing
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

If a path variable is of type `Uuid`, the import and the encoding is missing.

###### Output (without comment):

```elm
module Request.UserController exposing (get)

import Data.User as User exposing (User)
import Dict
import Http
import Json.Decode as Decode
import Url.Builder as Url


get :
    { onSend : Result Http.Error User -> msg
    , basePath : String
    , userId : Uuid
    }
    -> Cmd msg
get params =
    Http.request
        { method = "GET"
        , headers = []
        , url =
            Url.crossOrigin params.basePath
                [ "users", params.userId ]
                []
        , body = Http.emptyBody
        , expect = Http.expectJson params.onSend User.decoder
        , timeout = Just 30000
        , tracker = Nothing
        }
```

##### openapi-generator version

v4.0.0-beta2

##### OpenAPI declaration file content or url

```
    "/users/{userId}": {
      "get": {
        "tags": ["user-controller"],
        "summary": "get",
        "operationId": "get",
        "parameters": [
          {
            "name": "userId",
            "in": "path",
            "description": "userId",
            "required": true,
            "type": "string",
            "format": "uuid"
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": { "$ref": "#/definitions/User" }
          },
          "401": { "description": "Unauthorized" },
          "403": { "description": "Forbidden" },
          "404": { "description": "Not Found" }
        },
        "deprecated": false
      }
    }
```

##### Command line used for generation

```shell
docker run --rm -v /server-sdk:/local openapitools/openapi-generator-cli:v4.0.0-beta2 generate -i /local/swagger.json -o /local/sdk/elm -g elm --invoker-package abc --additional-properties elmEnableCustomBasePaths=true
```

##### Steps to reproduce

Build and look into output in Request files.

##### Related issues/PRs

This PR introduced UUIDs. https://github.com/OpenAPITools/openapi-generator/pull/1516

##### Suggest a fix

* add import
*  uuid to string with prm

This is what the output should look like (without comment)
```elm
module Request.UserController exposing (get)

-- CHANGED: Uuid import added manually

import Data.User as User exposing (User)
import Dict
import Http
import Json.Decode as Decode
import Url.Builder as Url
import Uuid exposing (Uuid)


get :
    { onSend : Result Http.Error User -> msg
    , basePath : String
    , userId : Uuid
    }
    -> Cmd msg
get params =
    Http.request
        { method = "GET"
        , headers = []
        , url =
            Url.crossOrigin params.basePath
                [ "users"
                , -- CHANGED: UUID.toString added manually
                  Uuid.toString params.userId
                ]
                []
        , body = Http.emptyBody
        , expect = Http.expectJson params.onSend User.decoder
        , timeout = Just 30000
        , tracker = Nothing
        }
```

### PR DESCRIPTION
**Pull request by eriktim**
And stringify UUIDs. Fixes #2484.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

* Add complex types dependencies in operations;
* Correctly stringify UUIDs.

cc @wing328

### CODE REVIEW

**Comment by andys8** [comment]
Wow, awesome! 😊

I can try out, if it works for our project, as soon as the fix is available with `openapitools/openapi-generator-cli:latest` on docker hub.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java [hunk 1]
```diff
@@ -474,6 +471,19 @@ private ElmImport createImport(final String name) {
...
+                || op.pathParams.stream().filter(param -> param.isDate).findAny().isPresent()
+                || op.headerParams.stream().filter(param -> param.isDate).findAny().isPresent();
+            hasDateTime = hasDate
+                || op.queryParams.stream().filter(param -> param.isDateTime).findAny().isPresent()
+                || op.pathParams.stream().filter(param -> param.isDateTime).findAny().isPresent()
+                || op.headerParams.stream().filter(param -> param.isDateTime).findAny().isPresent();
+            hasUuid = hasUuid
+                || op.queryParams.stream().filter(param -> param.isUuid).findAny().isPresent()
+                || op.pathParams.stream().filter(param -> param.isUuid).findAny().isPresent()
+                || op.headerParams.stream().filter(param -> param.isUuid).findAny().isPresent();
```
Comment (andys8) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java]: Not sure about the types in java, but you can probably combine all params (query, path, header) into a single collection or stream first, and filter afterwards only once, can you? ;)

Comment (eriktim) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java]: You're absolutely right this isn't very clean. I've updated the PR.

Comment (andys8) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java]: 👍

**Comment by andys8** [comment]
Could be merged :) @wing328

**Pull request merged**